assets/js/util/date-range.js

```diff
@@ -15,35 +15,36 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 /**
  * WordPress dependencies
  */
-import { applyFilters } from '@wordpress/hooks';
 import { __, _n, sprintf } from '@wordpress/i18n';
 
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+
 /**
  * Gets the current dateRange string.
  *
+ * @param {string} [slug] Optinal. The date range slug.
  * @return {string} the date range string.
  */
-export function getCurrentDateRange() {
-	/**
-	 * Filter the date range used for queries.
-	 *
-	 * @param String The selected date range. Default 'Last 28 days'.
-	 */
-	const dateRange = applyFilters( 'googlesitekit.dateRange', 'last-28-days' );
+export function getCurrentDateRange( slug ) {
+	const dateRange = slug || getCurrentDateRangeSlug();
 	const daysMatch = dateRange.match( /last-(\d+)-days/ );
 
 	if ( daysMatch && daysMatch[ 1 ] ) {
 		return sprintf(
 			/* translators: %s: Number of days matched. */
-			_n( '%s day', '%s days', parseInt( daysMatch[ 1 ], 10 ), 'google-site-kit' ),
+			_n( '%s day', '%s days', +daysMatch[ 1 ], 'google-site-kit' ),
```

Comment:
This is less characters, but please use `parseInt( num, 10 )` instead as it's more explicit and easier to see when reading.

Reply:
Updated.